src/ucp/core/ucp_request.h

```diff
@@ -251,6 +263,11 @@ struct ucp_recv_desc {
     uint32_t                payload_offset; /* Offset from end of the descriptor
                                              * to AM data */
     uint16_t                flags;          /* Flags */
+    int16_t                 priv_length;    /* Number of bytes consumed from
```

Comment:
Can be negative?

Reply:
yes, I use it as a negative with PTR_OFFSET macro

Reply:
do you use signed to avoid type cast?

Reply:
Yes, so I could use the following:
uct_iface_release_desc(UCS_PTR_BYTE_OFFSET(rdesc,
                                                                           -(UCP_WORKER_HEADROOM_PRIV_SIZE -
                                                                              rdesc->priv_length)));